packages/react-aria-components/src/Checkbox.tsx

```diff
@@ -159,10 +159,14 @@ function CheckboxGroup(props: CheckboxGroupProps, ref: ForwardedRef<HTMLDivEleme
 
 export const CheckboxContext = createContext<ContextValue<CheckboxProps, HTMLLabelElement>>(null);
 
-function Checkbox(props: CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
+function Checkbox(props: CheckboxProps & {inputRef?: RefObject<HTMLInputElement>}, ref: ForwardedRef<HTMLLabelElement>) {
   [props, ref] = useContextProps(props, ref, CheckboxContext);
-  let inputRef = useRef<HTMLInputElement>(null);
+  let {
+    inputRef: userInputRef
```

Comment:
this won't merge the inputRef from a user provided prop vs the context (I think useContextProps will only merge the ref in the second argument)
we could move this before the `useContextProps` call, then we'd be getting the user provided one, and then we could merge that with the one from the context
```
let {inputRef: userProvidedInputRef, ...otherProps} = props;
[props, ref] = useContextProps(otherProps, ref, CheckboxContext);
// now merge the refs userProvidedInputRef and props.inputRef
```

another option we have is to not expose an inputRef and instead useImperative to add focus/select to the forwardedRef
but I think the types for that would be more difficult